### PR TITLE
Fix needtable caption

### PR DESCRIPTION
**Pull request by haiyangToAI**
Fix for #348

### CODE REVIEW

File: sphinxcontrib/needs/directives/needtable.py [hunk 1]
```diff
@@ -64,16 +64,20 @@ def run(self):
 
         sort = self.options.get("sort", "id_complete")
 
-        caption = None
         if self.arguments:
-            caption = self.arguments[0]
+            title_text = self.arguments[0]
+            text_nodes = self.state.inline_text(title_text, self.lineno)[0]
+            title = nodes.title(title_text, "", *text_nodes)
```
Comment (danwos) [inline sphinxcontrib/needs/directives/needtable.py:70]: I would try to keep the creating of nodes out of run() and only collect data here.

The creation then is done in process().
So need_all_needtables mostly contain only str and not more

Comment (danwos) [reply sphinxcontrib/needs/directives/needtable.py:70]: So please move the node creation to process()

File: tests/test_needtable.py [hunk 2]
```diff
@@ -16,27 +16,28 @@ def test_doc_build_html(app, status, warning):
...
-# """
-#     assert figure in html
-#
-#     # negative test to check table without caption
-#     if sphinx.version_info[0] >= 4:
-#         assert '<table class="NEEDS_TABLE rtd-exclude-wy-table docutils align-default" id="needtable-index-1">'
-#         in html
-#     else:
-#         assert '<table class="NEEDS_TABLE rtd-exclude-wy-table docutils" id="needtable-index-1">' in html
+@with_app(buildername="xml", srcdir="doc_test/doc_needtable")
```
Comment (danwos) [inline tests/test_needtable.py:19]: you can/should keep html here.
But use ElementTree to analyse it.

Comment (danwos) [reply tests/test_needtable.py:19]: Or maybe leave it in the existing test-case (as it was before), but us elementree-checks instead of string-compares.

**Review by danwos — CHANGES_REQUESTED** [review]

File: sphinxcontrib/needs/directives/needtable.py (hunk 1 above)
Comment (haiyangToAI) [reply sphinxcontrib/needs/directives/needtable.py]: Done.

File: tests/test_needtable.py (hunk 2 above)
Comment (haiyangToAI) [reply tests/test_needtable.py]: Done.

File: pyproject.toml [hunk 3]
```diff
@@ -46,6 +46,7 @@ requests = "^2.25.1"
 [tool.poetry.dev-dependencies]
 
 # test dependencies
+lxml = ">=4.6.3"
```
Comment (danwos) [inline pyproject.toml:49]: Please use version selection as described here:
https://python-poetry.org/docs/dependency-specification/

In this case ``^4.6.3`` would be good

**Review by danwos — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (haiyangToAI) [reply pyproject.toml]: done

**Review by danwos — APPROVED** [review]

**Pull request merged**